### DISCUSSION TITLE
Debian sid : cannot convert to avif with vips "unsupported compression"

### DISCUSSION THREAD

**Discussion by dvhh** [libvips/libvips#3648] [Q&A]
Tested on 
OS: Debian sid
Arch: x86_64 and aarch64
Vips version: vips-8.14.4
Libvips: 8.14.4-1

When trying to convert from png to avif ( `vips copy input.png output.avif` ), I got an "unsupported compression", but am able to convert with heif-convert.

Already tried rebuilding libvips (with debugging flag), debugging `Unsupported compression` underlying heif seem to return heif_error_Unsupported_filetype/heif_suberror_Unspecified.

How would I debug or fix the issue ? Thanks for your help

**Comment by jcupitt** [comment]
Hi @dvhh,

I tried in docker:

```
$ docker run -it -v /home/john/pics:/data debian:sid
root@0628113fd8af:/# apt-get update
...
root@0628113fd8af:/# apt-get install libvips-tools --no-install-recommends
...
root@0628113fd8af:/data# vips copy k2.jpg /tmp/x.avif
heifsave: Unsupported compression
```

So I agree it looks like the libvips in debian sid has not been built with AVIF write support, although it does seem to be pulling in the relevant libraries.

I'll investigate.

**Reply by dvhh** [reply]
Thanks, I have investigate on libvips behavior only, will dig up to libheif with gdb

**Comment by kleisauke** [comment]
It looks like Debian builds the libheif dependencies as plugins after commit https://salsa.debian.org/multimedia-team/libheif/-/commit/e0e79b2d5348857604cadee4c33bc013eec5ce47. So, I believe this is the same issue as https://github.com/libvips/libvips/issues/3614.

**Reply by dvhh** [reply]
Thanks for the pointer, will get a look at it.

Edit: building with HEAD of libheif didn't seem to have fixed my issue, will dig further

**Reply by kleisauke** [reply]
> building with HEAD of libheif didn't seem to have fixed my issue, will dig further

Oh, curious. Are you able to test commit 52f9efad97188b6d98a0eb11621a68422188ede4? If that doesn't work either, then I'm not sure exactly what's going on.
